packages/service-provider-core/src/connect-info.ts

```diff
@@ -21,11 +18,59 @@ export interface ConnectionExtraInfo {
   node_version?: string;
   uri: string;
   is_local_atlas?: boolean;
+} & HostInformation;
+
+export type HostInformation = {
+  is_localhost?: boolean;
+  is_atlas_url?: boolean;
+  is_do_url?: boolean; // Is digital ocean url.
+};
+
+export function getHostnameForConnection(topology: any): string | undefined {
+  const resolvedHost =
+    topology?.s?.servers?.values().next().value.description.address ?? '';
```

Comment:
I know this is already part of the code here, but it's still worth asking – can we avoid this? `service-provider-core` isn't a place for code that depends on Node.js driver logic specifically (that's kinda the point of the package), and it would be great if we could avoid using internal properties like `.s` here

Reply:
Refactored to get topology from the topologyDescriptionChanged event.